lib/wrap-http-client.js

```diff
@@ -19,17 +19,18 @@ import createBackoff from './create-backoff'
  * @prop {boolean} retryOnTooManyRequests - If we should retry on 429s
  */
 export default function wrapHttpClient (http, {concurrency, delay, maxRetries, retryOnTooManyRequests}) {
-  return ['get', 'post', 'put', 'delete', 'patch', 'head'].reduce(
+  const backoffClient = ['get', 'post', 'put', 'delete', 'patch', 'head'].reduce(
     (http, methodName) => {
       let httpCall = http[methodName].bind(http)
       if (retryOnTooManyRequests) {
-        httpCall = maybeBackoff(httpCall, maxRetries)
+        http[methodName] = maybeBackoff(httpCall, maxRetries)
       }
-      http[methodName] = rateLimit(httpCall, concurrency, delay)
       return http
     },
     cloneDeep(http)
   )
+
+  return rateLimit(backoffClient, ['get', 'post', 'put', 'delete', 'patch', 'head'], concurrency, delay)
 }
 
 function maybeBackoff (fn, maxRetries) {
```

Comment:
With the current implementation we are handling errors (either `429` and `500`) twice. Once inside this function and after the `maxRetries` have been consumed here then we try again on this handler https://github.com/contentful/contentful-sdk-core/pull/13/files#diff-80d900d5fc6a583850c7b52862522c34R6

So I would remove the backoff logic and use only the `rateLimit` one. WDYT?

Another thing would be to remove the `retryOnTooManyRequests` option (and related code) as I would assume that most of the times we want to retry the requests. This of course is not core to the topic so could be done later.